utils/create-nestable-component.tsx

```
@@ -0,0 +1,24 @@
+import { ComponentType, createContext, useContext } from 'react';
+
+export function createNestableComponent<P extends object>(
+  Component: ComponentType<P>,
+): ComponentType<P> {
+  const PropsContext = createContext<P>({} as P);
+
+  const NestableComponent = (props: P) => {
+    const nestedProps = useContext(PropsContext);
+    const combinedProps = mergeDeep<P>({} as P, nestedProps, props);
+    const passableProps = { ...combinedProps };
+    if ('children' in passableProps) delete passableProps['children'];
```

Biome

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)